fe/fe-core/src/main/java/com/starrocks/scheduler/MVActiveChecker.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.scheduler;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.starrocks.alter.AlterJobMgr;
+import com.starrocks.analysis.TableName;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.MaterializedView;
+import com.starrocks.catalog.Table;
+import com.starrocks.common.Config;
+import com.starrocks.common.util.FrontendDaemon;
+import com.starrocks.qe.ConnectContext;
+import com.starrocks.server.GlobalStateMgr;
+import com.starrocks.statistic.StatisticUtils;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * A daemon thread that check the MV active status, try to activate the MV it's inactive.
+ */
+public class MVActiveChecker extends FrontendDaemon {
+
+    private static final Logger LOG = LogManager.getLogger(MVActiveChecker.class);
+
+    public MVActiveChecker() {
+        super("MVActiveChecker", Config.mv_active_checker_interval_seconds * 1000);
+    }
+
+    @Override
+    protected void runAfterCatalogReady() {
+        // reset if the interval has been changed
+        setInterval(Config.mv_active_checker_interval_seconds * 1000L);
+
+        try {
+            process();
+        } catch (Throwable e) {
+            LOG.warn("Failed to process one round of MVActiveChecker", e);
+        }
+    }
+
+    @VisibleForTesting
+    public void runForTest() {
+        process();
+    }
+
+    private void process() {
+        Collection<Database> dbs = GlobalStateMgr.getCurrentState().getIdToDb().values();
+        for (Database db : CollectionUtils.emptyIfNull(dbs)) {
+            for (Table table : CollectionUtils.emptyIfNull(db.getTables())) {
```

Comment:
what will happen if dropping the db during processing.

Reply:
1. the table list is got from a `ConcurrentHashMap`
2. `tryToActivate` will verify the existence of database again, throw exception if database doesn't exist